sklearn/ensemble/_hist_gradient_boosting/binning.py

```diff
@@ -29,31 +29,25 @@ def _find_binning_thresholds(data, max_bins, subsample, random_state):
         The maximum number of bins to use for non-missing values. If for a
         given feature the number of unique values is less than ``max_bins``,
         then those unique values will be used to compute the bin thresholds,
-        instead of the quantiles.
-    subsample : int or None
-        If ``n_samples > subsample``, then ``sub_samples`` samples will be
-        randomly chosen to compute the quantiles. If ``None``, the whole data
-        is used.
-    random_state: int, RandomState instance or None
-        Pseudo-random number generator to control the random sub-sampling.
-        Pass an int for reproducible output across multiple
-        function calls.
-        See :term: `Glossary <random_state>`.
+        instead of the quantiles
+    is_categorical : ndarray of bool or None
+        Indicates categorical features
 
     Return
     ------
-    binning_thresholds: list of arrays
+    binning_thresholds: list of arrays or None
```

Comment:
Should this be a dict too, for consistency?

`binning_thresholds` and `bin_categories` should ideally have the same type and the same function

Reply:
> Should this be a dict too, for consistency?

My ideal would be to combine them, since they are mutually exclusive:

```py
# threshold, is_numerical
thresholds = [(np.array[0, 2, 4], True), (np.array[2, 3], False)]
```

Is that too much?